### ISSUE TITLE
README documentation is very outdated

### ISSUE THREAD

**Issue by ekg** [vgteam/vg#305]
For example: https://github.com/vgteam/vg#mapping

```
If your graph is large, you want to use vg index to store the graph and vg map to align reads. vg map implements a kmer based seed and extend alignment model that is similar to that used in aligners like novoalign or MOSAIK. First an on-disk index is built with vg index which includes the graph itself and kmers of a particular size. When mapping, any kmer size shorter than that used in the index can be employed, and by default the mapper will decrease the kmer size to increase sensitivity when alignment at a particular k fails.
```

We now use xg/GCSA2, and map using maximal exact matches.

**Comment by edawson** [comment]
Lots of updates have happened to the wiki and README. However, this is a moving target and there is a lot of nuance to the command line incantations.

It would be good to decouple specifics from general usage. This is the goal of the [Quickstart](https://github.com/vgteam/vg/wiki/Quickstart) page, though we are still missing a proper manual page and our README probably has too much information still.

I think we should separate these three in addition to providing proper docs at a later date. I will keep working on basic usage, but we'll have to divide up the manual page.

**Comment by edawson** [comment]
A lot of the help messages have gotten changed in confusing ways. For example, with vg index:
  
  1. The `-d` option disappeared, but you can't create a Rocksdb index without it. 
  2. The `-D` option got relabeled. It used to describe it as a way to sort GAM, but now it just says "dump database contents to stdout." The problem is there's no mention of GAM sorting anywhere because of this change.
 
In my mind we'd streamline the CLI for user friendliness (e.g. GAM sorting probably belongs in the `sort` command and not `index`, even though the code is with indexing). This will lead to spaghetti code, but right now we have spaghetti usage.

What do the people think?

**Comment by faithokamoto** [comment]
I agree that it's annoying when all these pages are out of sync, but also:
- Having a single issue open about it isn't useful, since the problem is actually a bunch of little things that we play whack-a-mole with
- No one ever finds time to do this unless they want to
- Hello, I am the "they" in the previous bullet point

So, yes, I am trying to fix this stuff up, but I'm closing this issue. If there are any specific problems then they should be their own issues.
